CoreVulkanDisplay/src/au/gov/asd/tac/constellation/visual/vulkan/renderables/CVKIconsRenderable.java

```diff
@@ -956,381 +1460,627 @@ public int DestroyDescriptorPoolResources() {
+        // the staging buffer.  Rebuild also if we our vertex count has somehow changed.
+        final boolean rebuildRequired = cvkVertexFlagsStagingBuffer == null || vertexCount != access.getVertexCount();
+        if (rebuildRequired) {
+            RebuildVertexFlagsStagingBuffers(access, newVertexCount);
+        } else {
+            int minMax[] = change.getRange();
+            CVKAssert(minMax != null);
+            final int numChanges = change.getSize();
+            try {
+                vertexFlagsStagingBufferLock.lock();
+                
+                final int mappingStart = minMax[0] * FLAGS_STRIDE;
+                final int mappingSize  = ((minMax[1] - minMax[0]) + 1) * FLAGS_STRIDE;
+                ByteBuffer pVertexFlagsMemory = cvkVertexFlagsStagingBuffer.StartWrite(mappingStart, mappingSize);
+                for (int i = 0; i < numChanges; ++i) {
+                    int pos = change.getElement(i);
+                    SetVertexFlags(pos, minMax[0], pVertexFlagsMemory, access);                               
+                }
+                cvkVertexFlagsStagingBuffer.EndWrite();
+                pVertexFlagsMemory = null; // now unmapped, do not use
+            } finally {
+                vertexFlagsStagingBufferLock.unlock();
+            }   
+        }
+        
+        //=== EXECUTED BY RENDER THREAD (during CVKVisualProcessor.ProcessRenderTasks) ===//
+        return () -> {
+            if (rebuildRequired) {
+                vertexFlagsBufferState = CVK_RESOURCE_NEEDS_REBUILD;
+            } else if (vertexFlagsBufferState != CVK_RESOURCE_NEEDS_REBUILD) {
+                vertexFlagsBufferState = CVK_RESOURCE_NEEDS_UPDATE;
+            }
+        };         
+    }
+    
+    public CVKRenderableUpdateTask TaskUpdateColours(final VisualChange change, final VisualAccess access) {
+        //=== EXECUTED BY CALLING THREAD (VisualProcessor) ===//
+        CVKLOGGER.info(String.format("TaskUpdateColours frame %d: %d verts", parent.GetFrameNumber(), access.getVertexCount()));
+        
+        //=== EXECUTED BY RENDER THREAD (during CVKVisualProcessor.ProcessRenderTasks) ===//
+        return () -> {
+            if (vertexBuffersState != CVK_RESOURCE_NEEDS_REBUILD) {
+                vertexBuffersState = CVK_RESOURCE_NEEDS_UPDATE;
+            }
+        };             
+    }   
+    
+    public CVKRenderableUpdateTask TaskSetHighLightColour(final VisualAccess access) {
+        //=== EXECUTED BY CALLING THREAD (VisualProcessor) ===//
+        CVKLOGGER.info(String.format("TaskSetHighLightColour frame %d: %d verts", parent.GetFrameNumber(), access.getVertexCount()));
+        final ConstellationColor highlightColor = access.getHighlightColor();
+                                                
+        //=== EXECUTED BY RENDER THREAD (during CVKVisualProcessor.ProcessRenderTasks) ===//
+        return () -> {
+            geometryUBO.highlightColor.set(0, 0, highlightColor.getRed());
```

Comment:
It isn't possible for the Render thread to call this before the VisualProcessor is it?

Reply:
I'm not sure what you mean?

Reply:
Is it possible highlightColor is not created

Reply:
Oh I see.  Yeah true, we probably need a local cache this is updated in the Update of that UBO.  I will put that in a small fix after this PR is merged.

Reply:
Actually I am doing this now.